contrib/kube-prometheus/assets/prometheus/rules/general.rules

```diff
@@ -0,0 +1,109 @@
+### Up Alerting ###
+
+Alert TargetDown
+  IF up == 0
+  FOR 10m
+  LABELS {
+    severity = "warning"
+  }
+  ANNOTATIONS {
+    summary = "target is down",
+    description = "A target of type {{ $labels.job }} is down."
+  }
```

Comment:
This can potentially get very noisy as Cloud Native should all be about resilience to failure and this not being a symptom. Fine for now as a warning – especially for cluster components that don't scale to 100x+. But maybe add a TODO comment on whether this should be a fraction at some point, i.e. warn if more than 3% of targets for a job or `up == 0`.